src/hotspot/cpu/riscv/macroAssembler_riscv.hpp

```diff
@@ -678,6 +678,12 @@ class MacroAssembler: public Assembler {
   int pop_v(unsigned int bitset, Register stack);
 #endif // COMPILER2
 
+  bool in_range_auipc_s12(int64_t x) {
+    int64_t twoG = (2 * G);
+    int64_t twoK = (2 * K);
+    return x <= (twoG - twoK) && x >= (-twoG + twoK);
```

Comment:
As I remembered, the true range of RISC-V PC-relative addressing should be: [-2^31 - 2^11, 2^31 - 2^11)? See [1].

[1] https://patchwork.kernel.org/project/linux-riscv/patch/20220131182145.236005-3-kernel@esmil.dk/

Reply:
Ah, yes, thanks. I missed flipping and was thinking in semi-32-bit.
So the maximum auipc is 0x80..0 which is -2^31, if we subtract form this we would overflow on 32-bit.
But since we are subtracting from 64-bit register reach is actually larger than min int32.

I'll fix!